src/sched.rs

```diff
@@ -1,204 +1,116 @@
 mod cpuset_attribs {
-    use super::CpuMask;
-    pub const CPU_SETSIZE:           usize = 1024;
-    pub const CPU_MASK_BITS:         usize = 64;
-
-    #[inline]
-    pub fn set_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur | (1u64 << bit)
-    }
-
-    #[inline]
-    pub fn clear_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur & !(1u64 << bit)
-    }
+    pub const CPU_MASK_BITS: usize = 64;
+    pub const FLAG_BIT: u64 = 1;
 }
 
-#[cfg(all(target_arch = "x86", target_os = "linux"))]
+#[cfg(target_pointer_width = "32")]
 mod cpuset_attribs {
-    use super::CpuMask;
-    pub const CPU_SETSIZE:           usize = 1024;
-    pub const CPU_MASK_BITS:         usize = 32;
-
-    #[inline]
-    pub fn set_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur | (1u32 << bit)
-    }
-
-    #[inline]
-    pub fn clear_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur & !(1u32 << bit)
-    }
-}
-
-#[cfg(all(target_arch = "aarch64", any(target_os = "linux", target_os = "android")))]
-mod cpuset_attribs {
-    use super::CpuMask;
-    pub const CPU_SETSIZE:           usize = 1024;
-    pub const CPU_MASK_BITS:         usize = 64;
-
-    #[inline]
-    pub fn set_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur | (1u64 << bit)
-    }
-
-    #[inline]
-    pub fn clear_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur & !(1u64 << bit)
-    }
-}
-
-#[cfg(all(any(target_arch = "arm", target_arch = "mips"), target_os = "android"))]
-mod cpuset_attribs {
-    use super::CpuMask;
-    // bionic only supports up to 32 independent CPUs, instead of 1024.
-    pub const CPU_SETSIZE:          usize = 32;
-    pub const CPU_MASK_BITS:        usize = 32;
-
-    #[inline]
-    pub fn set_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur | (1u32 << bit)
-    }
-
-    #[inline]
-    pub fn clear_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur & !(1u32 << bit)
-    }
-}
-
-#[cfg(all(any(target_arch = "arm", target_arch = "mips"), target_os = "linux"))]
-mod cpuset_attribs {
-    use super::CpuMask;
-    pub const CPU_SETSIZE:          usize = 1024;
-    pub const CPU_MASK_BITS:        usize = 32;
-
-    #[inline]
-    pub fn set_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur | (1u32 << bit)
-    }
-
-    #[inline]
-    pub fn clear_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur & !(1u32 << bit)
-    }
+    pub const CPU_MASK_BITS: usize = 32;
+    pub const FLAG_BIT: u32 = 1;
 }
 
 pub type CloneCb<'a> = Box<FnMut() -> isize + 'a>;
 
 // A single CPU mask word
-pub type CpuMask = c_ulong;
+type CpuMask = c_ulong;
 
 // Structure representing the CPU set to apply
 #[repr(C)]
 #[derive(Clone, Copy)]
 pub struct CpuSet {
-    cpu_mask: [CpuMask; cpuset_attribs::CPU_SETSIZE/cpuset_attribs::CPU_MASK_BITS]
+    cpu_mask: [CpuMask; CPU_SETSIZE as usize / cpuset_attribs::CPU_MASK_BITS],
```

Comment:
When we use the`CPU_` functions (macros), the previous comment becomes obsolete.